### ISSUE TITLE
callback-error rule

### ISSUE THREAD

**Issue by xjamundx** [sindresorhus/eslint-plugin-unicorn#60]
I was wondering if we could have `no-callback-literal` rule to correspond with the `no-throw-literal` rule.

Basically, if you're using the callback pattern this should never be okay:

``` js
callback('error')
```

instead you should do something more like this:

``` js
callback(new Error('error'))
```

You can see the code here:
https://github.com/xjamundx/eslint-plugin-standard/pull/15

**Comment by busches** [comment]
So this rule already exists in `eslint-plugin-standard`? @sindresorhus where do you stand on duplicating rules that exist in other plugins already?

**Comment by xjamundx** [comment]
It does not, I proposed it to both. The standard team is still evaluating. I think it's a good idea though and we use it in our project locally.

**Comment by jfmengels** [comment]
Sounds like a good idea to me :+1:

**Comment by sindresorhus** [comment]
I don't think it should be duplicated here. I'm also lukewarm on it as I personally use promises over callbacks these days.

**Comment by jfmengels** [comment]
I agree that it should appear in only one plugin.

**Comment by xjamundx** [comment]
Thanks for the feedback. Agree on promises as well :)
